File: python/fin_ratios/types/market.py
"""Market and price data types."""
from __future__ import annotations
from dataclasses import dataclass
from typing import Optional


@dataclass
class MarketData:
    price: float
    market_cap: float
    shares_outstanding: Optional[float] = None
    enterprise_value: Optional[float] = None
    forward_eps: Optional[float] = None
    trailing_eps: Optional[float] = None
    dividend_per_share: Optional[float] = None
    high_52_week: Optional[float] = None
    low_52_week: Optional[float] = None
    ticker: Optional[str] = None
    as_of: Optional[str] = None


@dataclass
class PricePoint:
    date: str
    close: float
    volume: Optional[float] = None

File: typescript/src/types/market.ts
/**
 * Market and price data types.
 */

export interface MarketData {
  /** Current stock price */
  price: number
  /** Total market capitalization */
  marketCap: number
  /** Shares outstanding */
  sharesOutstanding?: number | null
  /** Enterprise value (if pre-calculated) */
  enterpriseValue?: number | null
  /** Analyst forward EPS estimate */
  forwardEps?: number | null
  /** Trailing 12-month EPS */
  trailingEps?: number | null
  /** Annual dividend per share */
  dividendPerShare?: number | null
  /** 52-week high */
  high52Week?: number | null
  /** 52-week low */
  low52Week?: number | null
  /** Ticker symbol */
  ticker?: string
  /** As-of date */
  asOf?: string
}

export interface PricePoint {
  date: string
  /** Adjusted close price */
  close: number
  /** Trading volume */
  volume?: number | null
}

export type PriceHistory = PricePoint[]

export interface Return {
  date: string
  /** Decimal return, e.g. 0.05 for 5% */
  value: number
}

export type ReturnSeries = Return[]
